plugins/check-path/core-rs/src/paths.rs

```diff
@@ -0,0 +1,84 @@
+use js_sys::Reflect;
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::rc::Rc;
+use wasm_bindgen::prelude::*;
+
+enum NodeValue {
+    StringType(String),
+    ObjectType(JsValue),
+    None,
+}
+
+pub struct Paths {
+    store: Rc<RefCell<HashMap<String, Vec<String>>>>,
+}
+
+impl Paths {
+    pub fn new() -> Self {
+        Self {
+            store: Rc::new(RefCell::new(HashMap::new())),
+        }
+    }
+
+    pub fn parse(&self, data: JsValue) {
+        self.traverse(data);
+    }
+
+    pub fn get(&self, key: &str) -> Vec<String> {
+        self.store
+            .borrow()
+            .get(key)
+            .or(Some(&vec!["".to_string()]))
```

Comment:
```
            .cloned()
            .unwrap_or_else(|| vec![String::new()])
```
and remove the clone at the end.
Does this need to return a vec with an empty string in it? Or is just an empty vec enough?

Reply:
empty string is expected in the consuming code - JS tool does that

Reply:
When I change the return type to a borrow of the `Vec<String>` I get the whole borrow of value owned in this function, I think I need some lifetime annotation or something here I think... idk yet, still lots to figure out in this file, but definitely will dig more into this.

Reply:
I think I improved this significantly by wrapping the store hash map in a reference counter pointer `Rc`